### ISSUE TITLE
UX improvement: Copying and pasting a single file in the Explorer should pop up a dialog to set the name

### ISSUE THREAD

**Issue by DanKaplanSES** [microsoft/vscode#159724]
**TL;DR:** Skip to the end of this issue to see my proposed solution.

When you copy a file in the Explorer and paste, it creates a file with the same name, suffixed with ` copy` then the extension. I've _never_ wanted to leave a file with this name, so I always have to rename it. From a usability perspective, this creates a lot of extra steps because after copying and pasting the file:

1. The cursor is focused inside the file editor instead of on the new file in the Explorer; this means I have to right click on the file and select `rename` every time. Or, for the more keyboard savvy, use a shortcut to focus the Explorer then the keyboard shortcut to rename the file.
2. (When you choose rename in the Explorer, it highlights the whole name, excluding the file extension.) I often want to modify the file name instead of completely change it, and removing that ` copy` from the end adds a bunch of extra key strokes to remove, particularly when I want to modify something at the beginning of the file name. 
    For example, let's say I want to copy a file named `foo-a-very-difficult-to-memorize-name.ts` to a file named `bar-a-very-difficult-to-memorize-name.ts`. Initially, VS code will name it `foo-a-very-difficult-to-memorize-name copy.ts` and highlight everything but the extension. From there I have two ways to get the name I want: 
    1. I remove the beginning, type `bar`, jump to the end, move the cursor to just before the extension, highlight the word `copy` with ctrl-shift-left, highlight the space with shift-left, then press delete, then press enter. 
    2. Alternatively, I can start at the end, but the process is a similar amount of steps. 
  Considering I only want to change three characters, the keystrokes required seem disproportional to the task.

Now, **as for a solution, here's what I recommend**: if you paste a file into the Explorer, a dialog should pop up asking you what file name you want. It should start with exactly the same name as the original filename, everything highlighted but the extension. An okay and cancel button should be on this dialog. Since there will already be a file with that name, the okay button will be disabled until you modify the file to be contextually valid.

The downside to this feature is a dialog will pop up if you paste a file into a different directory. If you do that right now, it'll paste the original filename, saving the step of pressing enter if a dialog popped up. But I believe this will reduce the number of keystrokes the majority of the time.

**Comment by lramos15** [comment]
There are no plans to address this at this time. I recommend changing your incremental name setting to smart. `"explorer.incrementalNaming": "smart"`. This will then cause `foo.js` to be pasted as `foo.1.js` you can then trigger rename and rename the file more to how you expect
